container-host-files/etc/scf/config/opinions.yml

```diff
@@ -48,100 +48,97 @@ properties:
     cf:
       client_id: app_autoscaler
       grant_type: client_credentials
-      password: "" # unused
-      username: "" # unused
+      skip_ssl_validation: true
```

Comment:
Why is this necessary?

Reply:
autoscaler will connect to uaa api for authorization and cf api to get app info and trigger scale.  When running with the self-signed certs ,   we need this flag to ensure the correctly connection .

Reply:
Maybe I'm confused, but doesn't this just mean that you are missing to authorize the internal CA in some of your roles?

```
  - name: authorize-internal-ca # needs to be second as it is a dependency of other jobs.
    release: scf
```

Assuming you include this job in all your roles, all self-signed certs should verify correctly.

Reply:
@jandubois ,  I need to keep 'skip_ssl_validation' here since the access target is the CC public API,  not for internal CA. Just like the "cf cli" and "smoke" test need to use "skip_ssl_valication" flag ..   If not,   autoscaler will report issue 
```
{"timestamp":"2019-06-06T05:01:53.211Z","source":"autoscaler:apiserver","text":"Failed to get Cloud Foundry API information","log_level":"error","data":{"info url":"https://api.cf.svc.cluster.local:9024/v2/info","error":{"code":"UNABLE_TO_VERIFY_LEAF_SIGNATURE","statusCode":500}}}
```